addons/mail/static/src/core/common/message.js

```diff
@@ -559,4 +577,17 @@ export class Message extends Component {
             message: this.props.message,
         });
     }
+
+    async onClickToggleTranslation() {
+        if (!this.message.translationValue) {
+            const { error, source_lang, body } = await this.rpc("/mail/message/translate", {
+                message_id: this.message.id,
+            });
+            this.message.translationValue = body && markup(body);
```

Comment:
In which case the translation would not return body?

Reply:
If an error occured in the controller, only the error is sent to the webclient. 
see: https://github.com/odoo/odoo/pull/126555/files#diff-4e5b3415df93c31414d3934312a85a875e63315066799b4c5314b00d55b80e0bR34